images/origin/scripts/recycler.sh

```diff
@@ -4,40 +4,31 @@
 # this script is intended to be used in a pod that performs the scrub.
 # the container in the pod should succeed or fail based on the exit status of this script.
 
-set -o errexit
 set -o nounset
 set -o pipefail
 
-function recycle(){
+# first and only arg is the directory to scrub
+dir=${1:-}
 
-    # first and only arg is the directory to scrub
-    dir=$1
-
-    if [ -z "$dir" ]; then
-        echo "Usage:  scrub_directory some/path/to/scrub"
-        return 1
-    fi
+if [ -z "${dir}" ]; then
+    echo "Usage:  recycler.sh some/path/to/scrub"
+    exit 1
+fi
 
-    if [ ! -e $dir ]; then
-        echo "scrub directory $dir does not exist"
-        return 1
-    fi
+echo "Scrubbing $dir"
 
-    # remove everything but keep the directory for re-use as a volume
-    rm -rfv $dir/*
+# only delete the volume's contents
+# retain the directory for re-use by another pod.
+rm -rfv $dir/*
 
-    if [ -z "$(ls -A $dir)" ]; then
-        echo "scrub directory $dir is empty"
-        return 0
-    else
-        echo "scrub directory $dir is not empty"
-        return 1
-    fi
-}
+rmExitCode=$?
 
-if recycle $1; then
+if [[ rmExitCode -eq 0 ]]; then
```

Comment:
@ncdc can you suggest another reviewer for this - I'm slammed and someone
with bash expertise would help

On Thu, Jul 30, 2015 at 10:23 AM, Mark Turansky <notifications@github.com>
wrote:

> In images/origin/scripts/recycler.sh
> <https://github.com/openshift/origin/pull/3657#discussion_r35875254>:
>
> >
> > -if recycle $1; then
> > +if [[ rmExitCode -eq 0 ]]; then
>
> fixed. I don't understand, though, why the script would work with and
> without the $. I think what's there should be wrong, as you observed.
>
> But I test the script running as root (write some files, invoke the
> script). The I su to another user and correctly get a permission denied
> error when using the script.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/3657/files#r35875254>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift